### ISSUE TITLE
Way to select certain fields from related tables for prisma2

### ISSUE THREAD

**Issue by P4trik** [prisma/prisma2#653]
Hello *,

if I query a table together with a related table, how can I select only **certain** fields of that **related** table?
```
const productsResult = await pt.products.findOne({
where: {
product_id: id
},
include:{
product_variants: true
}
});
```

In this case, I just want to have the "**SKU**" field of the "**product_variants**" table and the "**name**" field of "**products**".
Is that possible at all?

Thank's for answering!

**Comment by divyenduz** [comment]
Yes, that behavior is documented here: https://github.com/prisma/prisma2/blob/master/docs/relations.md#nested-reads-eager-loading

Does that answer your question?

**Comment by P4trik** [comment]
Thank's for your support!
Unfortunately not.
Setting a field selection leads to the following error message:

`const productsResult = await pt.products.findOne({
      where: {
        product_id: id
      },
      select: {
        name: true
      },
      include:{
        product_variants: true,
      }
    });`

Property 'name' does not exist on type '"Please either choose `select` or `include`"'.

But I want to select certain fields of products and/or product_variants to

**Comment by divyenduz** [comment]
@P4trik Can you please create a small reproduction with your schema file and the steps to reach this error?

**Comment by pantharshit00** [comment]
@P4trik 

You can achieve result using the following code:
```
const users = await pt.products.findOne({
	where: {
	  product_id: id
	},
    include: {
      product_variants: {
        select: {
          id: true,
          name: true, // or any other field you want from the relation
        },
      },
    },
  })
```

Feel free to close this if this solves your issue, otherwise please share your schema.

**Comment by P4trik** [comment]
@pantharshit00 Thats exactly what I'm looking for but unfortunately I receive that error:

![image](https://user-images.githubusercontent.com/2085861/66344358-f09bb280-e94d-11e9-8ddd-72991ffbd548.png)

**Comment by pantharshit00** [comment]
Can you please share your datamodel? Looks like I assumed the type of the product_variants incorrectly.

**Comment by P4trik** [comment]
`model products {
   product_id    Int    @id
   name String
   description String?
   product_variants product_variants[]
}

model product_variants {
    product_variants_id Int @id
    sku String? 
    product products @map(name: "product_id")
    category categories?
    descriptions String?
}
`

**Comment by pantharshit00** [comment]
Reproduction: https://github.com/harshit-test-org/prisma-prisma2-653

Ok, the following query works in JS:
```js
const users = await photon.products.findOne({
    where: {
      product_id: 1,
    },
    include: {
      product_variants: {
        select: {
          sku: true,
          descriptions: true,
        },
      },
    },
  })
```

Returns the following result:
![image](https://user-images.githubusercontent.com/22195362/66486798-e7832080-eac8-11e9-8b90-d8bea12b5430.png)


But the code itself is throwing type errors in TypeScript:
![image](https://user-images.githubusercontent.com/22195362/66487249-9e7f9c00-eac9-11e9-9836-66e188cff853.png)


That means the reported problem is type generation issue rather than a runtime issue. 

So for now, while we fix the typings, you can add a ts-ignore to get rid of type warning temporarily:
```ts
  const data = await photon.products.findOne({
    where: {
      product_id: 1,
    },
    include: {
      product_variants: {
        //@ts-ignore
        select: {
          sku: true,
          descriptions: true,
        },
      },
    },
  })
```
